### PR TITLE
641: Match on domain name only of website not full URL

### PR DESCRIPTION
**Pull request by mariaogryczak**

### CODE REVIEW

File: src/main/java/org/benetech/servicenet/util/UrlNormalizationUtils.java [hunk 1]
```diff
@@ -8,6 +8,7 @@
     private static final String HTTPS = "HTTPS://";
     private static final String HTTP = "HTTP://";
     private static final String TRAILING = "/";
+    private static final String[] DOMENS = new String[] {"WIKIPEDIA", "GOOGLE", "WIX", "YAHOO", "GUIDESTAR", "YELP"};
```
Comment (awalkowiak) [inline src/main/java/org/benetech/servicenet/util/UrlNormalizationUtils.java]: domains

File: src/main/java/org/benetech/servicenet/util/UrlNormalizationUtils.java [hunk 2]
```diff
@@ -27,7 +28,16 @@ public static String normalize(String url) {
         if (result.toUpperCase(Locale.ROOT).endsWith(TRAILING)) {
             result = result.substring(0, result.length() - 1);
         }
+        for (String domen:DOMENS) {
```
Comment (awalkowiak) [inline src/main/java/org/benetech/servicenet/util/UrlNormalizationUtils.java]: String domain: DOMAINS

File: src/main/webapp/app/modules/conflicts/shared/components/input-field.tsx [hunk 3]
```diff
@@ -225,6 +230,22 @@ export class InputField extends React.Component<IInputFieldProp, IInputFieldStat
       );
     }
 
+    let urlButton = null;
+    if (fieldName === 'url' && defaultValue) {
+      urlButton = (
+        <div className="url-icon" onClick={this.handleUrl(this.props.activity.organization.url)}>
```
Comment (awalkowiak) [inline src/main/webapp/app/modules/conflicts/shared/components/input-field.tsx]: Cant you just put FontAwesomeIcon inside anchor tag? Whats the point of having clickable div (which handles url) and anchor tag inside it?

File: src/main/webapp/i18n/en/multiRecordView.json [hunk 4]
```diff
@@ -15,6 +15,7 @@
       "error": "Error occurred while removing the match!"
     },
     "yourData": "Your data",
+    "clickHereForWebsite": "Click the icon for website",
```
Comment (awalkowiak) [inline src/main/webapp/i18n/en/multiRecordView.json]: This sounds a bit weird. What icon, the question mark one? What do you mean by "for website"? Will it be copied to clipboard or will user be redirected to it?

File: src/main/java/org/benetech/servicenet/util/UrlNormalizationUtils.java [hunk 5]
```diff
@@ -27,7 +28,16 @@ public static String normalize(String url) {
...
+            if ((result.toUpperCase(Locale.ROOT).contains(domen))) {
+                return result;
+            }
+        }
+        if (result.toUpperCase(Locale.ROOT).contains(TRAILING)) {
```
Comment (awalkowiak) [inline src/main/java/org/benetech/servicenet/util/UrlNormalizationUtils.java]: Should not this part be before domain check?

Comment (pmuchowski) [reply src/main/java/org/benetech/servicenet/util/UrlNormalizationUtils.java]: I think not, because for those few domains you want to return the exact url

Comment (mariaogryczak) [reply src/main/java/org/benetech/servicenet/util/UrlNormalizationUtils.java]: Yes, as @pmuchowski wrote

File: src/main/webapp/i18n/en/multiRecordView.json (hunk 4 above)
Comment (mariaogryczak) [reply src/main/webapp/i18n/en/multiRecordView.json]: User will be redirected. Would 'Go to website' be better?

File: src/main/java/org/benetech/servicenet/util/UrlNormalizationUtils.java (hunk 2 above)
Comment (pmuchowski) [inline src/main/java/org/benetech/servicenet/util/UrlNormalizationUtils.java]: there should be a whitespace on both sides of the `:`

File: src/main/java/org/benetech/servicenet/util/UrlNormalizationUtils.java (hunk 5 above)
Comment (awalkowiak) [reply src/main/java/org/benetech/servicenet/util/UrlNormalizationUtils.java]: ok, cool

File: src/main/webapp/app/modules/conflicts/shared/components/input-field.tsx [hunk 6]
```diff
@@ -5,6 +5,7 @@ import classnames from 'classnames';
 import { connect } from 'react-redux';
 import { Translate, TextFormat, translate } from 'react-jhipster';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faExternalLinkAlt } from '@fortawesome/free-solid-svg-icons';
```
Comment (awalkowiak) [inline src/main/webapp/app/modules/conflicts/shared/components/input-field.tsx]: This should be imported in icon-loader.

File: src/main/webapp/app/modules/conflicts/shared/components/input-field.tsx [hunk 7]
```diff
@@ -110,6 +111,10 @@ export class InputField extends React.Component<IInputFieldProp, IInputFieldStat
     document.body.removeChild(textArea);
   };
 
+  handleUrl = url => () => {
```
Comment (awalkowiak) [inline src/main/webapp/app/modules/conflicts/shared/components/input-field.tsx]: Are you still using this?

**Pull request merged**